src/App.test.js

```diff
@@ -209,4 +210,53 @@ describe("Advanced tests of app ", () => {
       .toJSON();
     expect(tree).toMatchSnapshot();
   });
+  describe("Show error messages", () => {
+    it("Show error message for incorrect type", () => {
+      let inputValue = '{"label":"123", "type":"incorrectType"}';
+      const store = createStore(reducer);
+      store.dispatch(openOverlay());
+      store.dispatch(setValues(inputValue));
+
+      const tree = renderer
+        .create(
+          <Provider store={store}>
+            <Input />
+          </Provider>
+        )
+        .toJSON();
+      expect(tree).toMatchSnapshot();
+    });
+    it("Show error message for incorrect options", () => {
+      let inputValue =
+        '"{"label":"12","type":"select","incorrectOptions":["12","12","12"]}"';
+      const store = createStore(reducer);
+      store.dispatch(openOverlay());
+      store.dispatch(setValues(inputValue));
+
+      const tree = renderer
+        .create(
+          <Provider store={store}>
+            <Input />
+          </Provider>
+        )
+        .toJSON();
+      expect(tree).toMatchSnapshot();
```

Comment:
This shows the message for incorrect JSON and not for incorrect options.

Reply:
Okay, I will check that.